core/src/main/scala/kafka/tools/ConsumerPerformance.scala

```diff
@@ -210,7 +215,10 @@ object ConsumerPerformance {
    
     val useNewConsumer = options.has(useNewConsumerOpt)
     
-    val props = new Properties
+    val props = if (options.has(consumerConfigOpt))
+      Utils.loadProps(options.valueOf(consumerConfigOpt))
```

Comment:
Agreed, ideally if they were specified they would override, but if unspecified it would use values from the file. Not sure how easy that is to get working with this options library. I could have sworn I checked one of the other tools and found it using the behavior I expected, but now can't find it. I don't think this is make-or-break. I was just raising it since the behavior is unintuitive to me and although the performance tools are less critical, we probably wouldn't want to change their behavior that significantly after releasing a version with support for the `consumer.config` option.